packages/tests/src.ts/test-contract.spec.ts

```diff
@@ -306,8 +307,28 @@ describe("Test Contract Transaction Population", function() {
         assert.strictEqual(Array.isArray(transactions), true);
         assert.strictEqual(transactions.length, 2);
     });
+
+    // TODO: skipped as we should not spam testnet with random contracts
+    // Previewnet is not really a choice as it will (soon or later) result in INVALID_SIGNATURE pre-checks as of cleanup
+    xit("should be able to deploy a contract", async function() {
+        const hederaEoa = {
+            account: "0.0.29559509",
+            privateKey: "0xbb621d187c22459ab6ed6768bd516bd630a087df4d5a4fbe95d77e87af10c6e1"
+        };
+        const provider = ethers.providers.getDefaultProvider('testnet');
+        // @ts-ignore
+        const wallet = new ethers.Wallet(hederaEoa, provider);
+        const abi = JSON.parse(fs.readFileSync('examples/assets/abi/GLDToken_abi.json').toString());
+        const bytecode = fs.readFileSync('examples/assets/bytecode/GLDToken.bin').toString();
+        const contractFactory = new ethers.ContractFactory(abi, bytecode, wallet);
+        const contract = await contractFactory.deploy();
```

Comment:
```TypeScript
const contract = await contractFactory.deploy();
const deployTx = await contract.wait();
assert(deployTx.contractAddress)
```

Reply:
`deploy` returns a contract instance right now, which doesn't have a `wait()` method. We can await the deploymentTransaction (`contract.deployTransaction`), but it's already a resolved promise. Am I missing something?